linera-sdk/src/abis/fungible.rs

```diff
@@ -0,0 +1,253 @@
+        /// Amount to be transferred
+        amount: Amount,
+        /// Target account to transfer the amount to
+        target_account: Account,
+    },
+    /// Same as `Transfer` but the source account may be remote. Depending on its
+    /// configuration, the target chain may take time or refuse to process
+    /// the message.
+    Claim {
+        /// Source account to claim amount from
+        source_account: Account,
+        /// Amount to be claimed
+        amount: Amount,
+        /// Target account to claim the amount into
+        target_account: Account,
+    },
+}
+
+/// A fungible response
+#[derive(Debug, Deserialize, Serialize, Default)]
+pub enum FungibleResponse {
+    /// Ok response
+    #[default]
+    Ok,
+    /// Balance response
+    Balance(Amount),
+    /// Ticker symbol response
+    TickerSymbol(String),
+}
+
+/// The initial state to initialize fungible with
+#[derive(Clone, Debug, Deserialize, Eq, Ord, PartialEq, PartialOrd, Serialize)]
+pub struct InitialState {
+    /// Accounts and their respective initial balances
+    pub accounts: BTreeMap<AccountOwner, Amount>,
+}
+
+/// The parameters to initialize fungible with
+#[derive(Clone, Debug, Deserialize, Eq, Ord, PartialEq, PartialOrd, Serialize)]
+pub struct Parameters {
+    /// Ticker symbol for the fungible
+    pub ticker_symbol: String,
+}
+
+impl Parameters {
+    /// Initialize parameters
+    pub fn new(ticker_symbol: &str) -> Self {
+        let ticker_symbol = ticker_symbol.to_string();
+        Self { ticker_symbol }
+    }
+}
+
+/// An account.
+#[derive(
+    Clone,
+    Copy,
+    Debug,
+    Deserialize,
+    Eq,
+    Ord,
+    PartialEq,
+    PartialOrd,
+    Serialize,
+    SimpleObject,
+    InputObject,
+)]
+pub struct Account {
+    /// Chain ID of the account
+    pub chain_id: ChainId,
+    /// Owner of the account
+    pub owner: AccountOwner,
+}
+
+/// A builder type for constructing the initial state of the application.
+#[derive(Debug, Default)]
+pub struct InitialStateBuilder {
+    /// Accounts and their respective initial balances
+    account_balances: BTreeMap<AccountOwner, Amount>,
+}
+
+impl InitialStateBuilder {
+    /// Adds an account to the initial state of the application.
+    pub fn with_account(mut self, account: AccountOwner, balance: impl Into<Amount>) -> Self {
+        self.account_balances.insert(account, balance.into());
+        self
+    }
+
+    /// Returns the serialized initial state of the application, ready to used as the
+    /// initialization argument.
+    pub fn build(&self) -> InitialState {
+        InitialState {
+            accounts: self.account_balances.clone(),
+        }
+    }
+}
+
+#[cfg(all(
+    feature = "test",
+    any(feature = "wasmer", feature = "wasmtime"),
+    not(target_arch = "wasm32")
+))]
+impl FungibleTokenAbi {
+    /// Creates a fungible token application and distributes `initial_amounts` to new individual
+    /// chains.
+    pub async fn create_with_accounts(
```

Comment:
(This is only useful for application tests anyway, is it? I'd keep this outside `linera-sdk`, too, but I'm not sure; it's not a blocker.)

Reply:
Ha good point, this is what I was fighting CI with 🤣
I'll keep it in `fungible`

Reply:
Ah actually I can't because `struct FungibleTokenAbi` will be defined here now 😅

Reply:
But it doesn't have to be a method of that; it could just be a free-standing function for now?

Reply:
+1

In doubt, keep unstable APIs private and/or test-only.

Reply:
Ah good point